mlflow/telemetry/track.py

```diff
@@ -2,29 +2,28 @@
 import inspect
 import logging
 import time
-from typing import Any, Callable, Optional
+from typing import Any, Callable, Optional, ParamSpec, TypeVar
 
 from mlflow.telemetry.client import get_telemetry_client
 from mlflow.telemetry.parser import API_PARSER_MAPPING
 from mlflow.telemetry.schemas import APIRecord, APIStatus
 from mlflow.telemetry.utils import (
     _disable_telemetry,
-    _disable_telemetry_tracking_var,
-    invoked_from_internal_api,
     is_telemetry_disabled,
+    should_skip_telemetry,
 )
 
 _logger = logging.getLogger(__name__)
 
 
-def track_api_usage(func: Callable[..., Any]) -> Callable[..., Any]:
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def track_api_usage(func: Callable[P, R]) -> Callable[P, R]:
```

Comment:
```suggestion
def track_api_usage(func: Callable[P, R]) -> Callable[P, R]:
    if is_telemetry_disabled():
        return func
```

In a a follow-up, can we directly return `func` if `is_telemetry_disabled()` is to remove an unnecessary call stack?

Reply:
Yes make sense